src/MCPServer/lib/archivematicaMCP.py

```diff
@@ -260,6 +260,74 @@ def cleanupOldDbEntriesOnNewRun():
 
 
+def created_shared_directory_struture():
+    dirs = (
+        "arrange",
+        "completed",
+        "completed/transfers",
+        "currentlyProcessing",
+        "DIPbackups",
+        "failed",
+        "rejected",
+        "sharedMicroServiceTasksConfigs",
+        "sharedMicroServiceTasksConfigs/createXmlEventsAssist",
+        "sharedMicroServiceTasksConfigs/generateAIP",
+        "sharedMicroServiceTasksConfigs/generateAIP/bagit",
+        "sharedMicroServiceTasksConfigs/processingMCPConfigs",
+        "sharedMicroServiceTasksConfigs/transcoder",
+        "sharedMicroServiceTasksConfigs/transcoder/defaultIcons",
+        "SIPbackups",
+        "tmp",
+        "watchedDirectories",
+        "watchedDirectories/activeTransfers",
+        "watchedDirectories/activeTransfers/baggitDirectory",
+        "watchedDirectories/activeTransfers/baggitZippedDirectory",
+        "watchedDirectories/activeTransfers/Dspace",
+        "watchedDirectories/activeTransfers/maildir",
+        "watchedDirectories/activeTransfers/standardTransfer",
+        "watchedDirectories/activeTransfers/TRIM",
+        "watchedDirectories/approveNormalization",
+        "watchedDirectories/approveSubmissionDocumentationIngest",
+        "watchedDirectories/quarantined",
+        "watchedDirectories/SIPCreation",
+        "watchedDirectories/SIPCreation/completedTransfers",
+        "watchedDirectories/SIPCreation/SIPsUnderConstruction",
+        "watchedDirectories/storeAIP",
+        "watchedDirectories/system",
+        "watchedDirectories/system/autoProcessSIP",
+        "watchedDirectories/system/autoRestructureForCompliance",
+        "watchedDirectories/system/createAIC",
+        "watchedDirectories/system/reingestAIP",
+        "watchedDirectories/uploadDIP",
+        "watchedDirectories/uploadedDIPs",
+        "watchedDirectories/watchedDirectories",
+        "watchedDirectories/watchedDirectories/system",
+        "watchedDirectories/watchedDirectories/system/autoProcessSIP",
+        "watchedDirectories/workFlowDecisions",
+        "watchedDirectories/workFlowDecisions/compressionAIPDecisions",
+        "watchedDirectories/workFlowDecisions/createDip",
+        "watchedDirectories/workFlowDecisions/createTree",
+        "watchedDirectories/workFlowDecisions/examineContentsChoice",
+        "watchedDirectories/workFlowDecisions/extractPackagesChoice",
+        "watchedDirectories/workFlowDecisions/metadataReminder",
+        "watchedDirectories/workFlowDecisions/quarantineTransfer",
+        "watchedDirectories/workFlowDecisions/selectFormatIDToolIngest",
+        "watchedDirectories/workFlowDecisions/selectFormatIDToolTransfer",
+        "www",
+        "www/AIPsStore",
+        "www/AIPsStore/transferBacklog",
+        "www/AIPsStore/transferBacklog/arrange",
+        "www/AIPsStore/transferBacklog/originals",
+        "www/DIPsStore"
+    )
+    for d in dirs:
+        d = os.path.join(django_settings.SHARED_DIRECTORY, d)
```

Comment:
I don't pretend to know a lot about the MCP or SS, but avoiding a single component doing all of the set up makes sense to me.

I think each component should be responsible for creating the dirs that it is the "producer" for - like a producer generally creates the queue for messages in a messaging system. If a "consumer", i.e. a process wishing to use files in the sharedDirectory finds that location missing then it can (probably) just assume that there's nothing there for it to use.

Again, I may be speaking out of turn here so ignore me if I'm talking rubbish. :)